### PR TITLE
Ungate global_fetch_strictly_public compatibility flag

### PR DESCRIPTION
**Pull request by irvinebroque**
Docs: https://github.com/cloudflare/cloudflare-docs/pull/18075

### CODE REVIEW

File: src/workerd/io/compatibility-date.capnp [hunk 1]
```diff
@@ -501,7 +501,6 @@ struct CompatibilityFlags @0x8f8c1b68151b6cef {
   globalFetchStrictlyPublic @51 :Bool
       $compatEnableFlag("global_fetch_strictly_public")
       $compatDisableFlag("global_fetch_private_origin")
```
Comment (irvinebroque) [inline src/workerd/io/compatibility-date.capnp:503]:
```suggestion
      $compatDisableFlag("global_fetch_private_origin");
```

**Comment by irvinebroque** [comment]
@mikenomitch @kentonv @jasnell @danlapid FYI

**Review by danlapid — APPROVED** [review]
🤩

**Review by anonrig — APPROVED** [review]

File: src/workerd/io/compatibility-date.capnp [hunk 2]
```diff
@@ -500,8 +500,7 @@ struct CompatibilityFlags @0x8f8c1b68151b6cef {
 
   globalFetchStrictlyPublic @51 :Bool
       $compatEnableFlag("global_fetch_strictly_public")
-      $compatDisableFlag("global_fetch_private_origin")
-      $experimental;
+      $compatDisableFlag("global_fetch_private_origin");
```
Comment (jasnell) [inline src/workerd/io/compatibility-date.capnp:503]: Just want to confirm that we do not intend for this to become default on with some specific compat date?

**Review by jasnell — APPROVED** [review]

File: src/workerd/io/compatibility-date.capnp (hunk 2 above)
Comment (danlapid) [reply src/workerd/io/compatibility-date.capnp:503]: Currently - no
In the future - probably 😅

**Pull request merged**